packages/uni_ui/lib/cards/exam_card.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:flutter/material.dart';
+import 'package:phosphor_flutter/phosphor_flutter.dart';
+import 'package:uni_ui/cards/generic_card.dart';
+import 'package:uni_ui/theme.dart';
+
+class ExamCard extends StatelessWidget {
+  const ExamCard({
+    super.key,
+    required this.name,
+    required this.acronym,
+    required this.rooms,
+    required this.type,
+    this.startTime,
+    this.isInvisible = false,
+    this.showIcon = true,
+    this.iconAction,
+  });
+
+  final String name;
+  final String acronym;
+  final List<String> rooms;
+  final String type;
+  final String? startTime;
+  final bool isInvisible;
+  final bool showIcon;
+  final Function()? iconAction;
+
+  @override
+  Widget build(BuildContext context) {
+    return Opacity(
+      opacity: isInvisible ? 0.6 : 1.0,
+      child: GenericCard(
+        key: key,
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            Expanded(
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Row(
+                    children: [
+                      Text(
+                        acronym,
+                        overflow: TextOverflow.ellipsis,
+                        style: Theme.of(context).textTheme.headlineMedium!,
+                      ),
+                      const SizedBox(width: 8),
+                      Badge(
+                        label: Text(type),
+                        backgroundColor: BadgeColors.er,
+                        textColor: Theme.of(context).colorScheme.surface,
+                      ),
+                    ],
+                  ),
+                  Text(
+                    name,
+                    overflow: TextOverflow.ellipsis,
+                    style: Theme.of(context).textTheme.titleLarge!,
+                  ),
+                  const SizedBox(height: 5),
+                  Row(
+                    children: [
+                      PhosphorIcon(
+                        PhosphorIcons.clock(PhosphorIconsStyle.duotone),
+                        color: Theme.of(context).iconTheme.color,
+                        size: 20,
+                      ),
+                      const SizedBox(width: 4),
+                      Text(
+                        startTime ?? "--:--",
+                        style: Theme.of(context).textTheme.bodyMedium!,
+                      ),
+                      const SizedBox(width: 8),
+                      if (!rooms.isEmpty)
+                        PhosphorIcon(
+                          PhosphorIcons.mapPin(PhosphorIconsStyle.duotone),
+                          color: Theme.of(context).iconTheme.color,
+                          size: 20,
+                        ),
+                      const SizedBox(width: 4),
+                      SizedBox(
+                        height: 20,
+                        child: ListView.separated(
```

Comment:
- To avoid potential overflow issues, consider wrapping the `SizedBox` with an `Expanded` or `Flexible` widget.
- Using `ListView.separated` with `NeverScrollableScrollPhysics` will hide the rooms that don't fit in the available space (add horizontal scroll or use `Wrap` widget, instead?)